### PR TITLE
1rmlastcalculatecount

### PR DESCRIPTION
**Pull request by Malekai**

### CODE REVIEW

File: app/redux/reducers/AnalysisReducer.js [hunk 1]
```diff
@@ -177,6 +180,10 @@ const AnalysisReducer = (state = defaultState, action) => {
                 dragged: !state.dragged,
             };
         case CALC_1RM:
+            let e1RMCalcs = state.e1RMCalcs;
+            let e1RMCalcExercise = action.e1RMCalc.exercise;
+            e1RMCalcs[e1RMCalcExercise] = action.e1RMCalc;
```
Comment (JonnyLinja) [inline app/redux/reducers/AnalysisReducer.js]: - [ ] have this save it per exercise AND tag combination

File: app/redux/sagas/InitializedAnalyticsSaga.js [hunk 2]
```diff
@@ -18,9 +20,17 @@ const InitializedAnalyticsSaga = function * InitializedAnalyticsSaga() {
...
         var numWorkoutSets = SetsSelectors.getNumWorkoutSets(state);
     }
+
+    const e1RMLastCalcs = AnalysisSelectors.getE1RMCalcs(state);
+    const e1RMLastCalcCount = AnalysisSelectors.getE1RMCalcsCount(state);
+
     Analytics.logEventWithAppState('initialized', {
         num_workout_sets: numWorkoutSets,
+        one_rm_last_calculate_count: e1RMLastCalcCount,
+        one_rm_last_calculate: e1RMLastCalcs,
```
Comment (JonnyLinja) [inline app/redux/sagas/InitializedAnalyticsSaga.js]: - [ ] it's not supposed to be a parameter on an event, it is ACTUALLY an event by itself that is run after STORE_INITIALIZED is run

File: app/redux/sagas/OneRMLastCalculateAnalyticsSaga.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+import {
+    STORE_INITIALIZED,
+} from 'app/configs+constants/ActionTypes';
+
+const OneRMLastCalculateAnalyticsSaga = function * OneRMLastCalculateAnalyticsSaga() {
+    yield take(STORE_INITIALIZED);
+    const state = yield select();
+
+    const e1RMLastCalcs = AnalysisSelectors.getE1RMCalcs(state);
+    const e1RMLastCalcCount = AnalysisSelectors.getE1RMCalcsCount(state);
```
Comment (JonnyLinja) [inline app/redux/sagas/OneRMLastCalculateAnalyticsSaga.js]: - [ ] What is the point of calcs count?

Comment (Malekai) [reply app/redux/sagas/OneRMLastCalculateAnalyticsSaga.js]: removed

File: app/features/analysis/calculate/OneRMCalculateActions.js [hunk 4]
```diff
@@ -173,8 +182,12 @@ const logCalculate1RMAnalytics = (state, exercise, numIncludeTags, numExcludeTag
         min_velocity: minVelocity,
         max_velocity: maxVelocity,
         velocity_range: maxVelocity - minVelocity,
-        metric: metric
-    }, state);
+        metric: metric,
+    }
```
Comment (JonnyLinja) [inline app/features/analysis/calculate/OneRMCalculateActions.js]: - [ ] semicolon

File: app/redux/reducers/AnalysisReducer.js [hunk 5]
```diff
@@ -191,8 +202,13 @@ const AnalysisReducer = (state = defaultState, action) => {
...
                 isRegressionNegative: action.isRegressionNegative,
                 scroll: !state.scroll,
+                e1RMCalcs: [ ...filteredCalcs, newCalc ],
             };
-        
+        case CLEAR_E1RM_CALCS:
+            return {
+                ...state,
+                e1RMCalcs: [],
+            }
```
Comment (Malekai) [inline app/redux/reducers/AnalysisReducer.js]: Semi

File: app/redux/reducers/AnalysisReducer.js [hunk 6]
```diff
@@ -304,4 +320,22 @@ const AnalysisReducer = (state = defaultState, action) => {
...
+
+}
+
+const areArraysEqual = (array1, array2) => {
+    if (!array1 || !array2 || array1.length !== array2.length) {
+        return false;
+    }
+    
+    return _.isEqual(array1.sort(), array2.sort());
+}
```
Comment (Malekai) [inline app/redux/reducers/AnalysisReducer.js]: Semi

File: app/redux/shared_actions/AnalysisActionCreators.js [hunk 7]
```diff
@@ -7,6 +7,7 @@ import {
     PRESENT_ALGORITHM,
     PRESENT_BEST_RESULTS,
     ANALYSIS_DRAGGED,
+    CLEAR_E1RM_CALCS,
```
Comment (JonnyLinja) [inline app/redux/shared_actions/AnalysisActionCreators.js]: - [ ] CLEAR_1RM_ANALYTICS

File: app/redux/shared_actions/AnalysisActionCreators.js [hunk 8]
```diff
@@ -46,3 +47,7 @@ const logBestResultsAnalytics = (state) => {
     Analytics.logEventWithAppState('one_rm_best_results', {
     }, state);
 };
+
+export const clearE1RMCalcs = () => ({
```
Comment (JonnyLinja) [inline app/redux/shared_actions/AnalysisActionCreators.js]: - [ ] clear1RMAnalytics

File: app/redux/selectors/AnalysisSelectors.js [hunk 9]
```diff
@@ -24,6 +24,8 @@ export const getVelocitySlider = (state) => stateRoot(state).velocitySlider;
 
 export const getDaysRange = (state) => stateRoot(state).daysRange;
 
+export const getE1RMCalcs = (state) => stateRoot(state).e1RMCalcs;
```
Comment (JonnyLinja) [inline app/redux/selectors/AnalysisSelectors.js]: - [ ] get1RMAnalytics

File: app/redux/sagas/Sagas.js [hunk 10]
```diff
@@ -11,6 +11,7 @@ import EndOldWorkoutSaga from './EndOldWorkoutSaga';
 import ReconnectSaga from './ReconnectSaga';
 import SurveySaga from './SurveySaga';
 import InitializedAnalyticsSaga from './InitializedAnalyticsSaga';
+import OneRMLastCalculateSaga from './OneRMLastCalculateAnalyticsSaga';
```
Comment (JonnyLinja) [inline app/redux/sagas/Sagas.js]: - [ ] OneRMAnalyticsSaga

File: app/redux/sagas/OneRMLastCalculateAnalyticsSaga.js [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+        max_velocity: maxVelocity,
+        velocity_range: velocityRange,
+        metric: metric
+    }, state);
+};
+
+// TODO: write a unit test for this
+// this is waiting for a system to test sagas
+
+export default OneRMLastCalculateAnalyticsSaga;
```
Comment (JonnyLinja) [inline app/redux/sagas/OneRMLastCalculateAnalyticsSaga.js]: - [ ] Rename entire thing to OneRMAnalyticsSaga

File: app/redux/reducers/AnalysisReducer.js [hunk 12]
```diff
@@ -68,6 +70,7 @@ const defaultState = {
     maxX: null,
     maxY: null,
     isRegressionNegative: null,
+    e1RMCalcs: [],
```
Comment (JonnyLinja) [inline app/redux/reducers/AnalysisReducer.js]: - [ ] oneRMAnalytics

File: app/redux/Store.js [hunk 13]
```diff
@@ -67,7 +67,7 @@ export default initializeStore = () => {
             createFilter('collapsedSettings', ['metric1', 'quantifier1', 'metric2', 'quantifier2', 'metric3', 'quantifier3', 'metric4', 'quantifier4', 'metric5', 'quantifier5']),            
             createFilter('workout', ['removedCounter', 'restoredCounter']),
             createFilter('history', ['viewedCounter']),
-            createFilter('analysis', ['exercise', 'velocitySlider', 'daysRange', 'tagsToInclude', 'tagsToExclude']),
+            createFilter('analysis', ['exercise', 'velocitySlider', 'daysRange', 'tagsToInclude', 'tagsToExclude', 'e1RMCalcs']),
```
Comment (JonnyLinja) [inline app/redux/Store.js]: - [ ] oneRMAnalytics

File: app/features/analysis/calculate/OneRMCalculateActions.js [hunk 14]
```diff
@@ -113,6 +121,7 @@ export const calcE1RM = () => (dispatch, getState) => {
         maxX: results.maxX,
         maxY: results.maxY,
         isRegressionNegative: results.isRegressionNegative,
+        e1RMCalc: e1RMCalcParams,
```
Comment (JonnyLinja) [inline app/features/analysis/calculate/OneRMCalculateActions.js]: - [ ] oneRMAnalytics

File: app/features/analysis/calculate/OneRMCalculateActions.js [hunk 15]
```diff
@@ -96,9 +96,17 @@ export const calcE1RM = () => (dispatch, getState) => {
         results.maxX,
         results.minY,
         results.maxY,
-        metric
+        metric,
     );
 
+    const e1RMCalcParams = {
```
Comment (JonnyLinja) [inline app/features/analysis/calculate/OneRMCalculateActions.js]: - [ ] storedAnalyticsParams

File: app/features/analysis/calculate/OneRMCalculateActions.js [hunk 16]
```diff
@@ -78,8 +78,8 @@ export const calcE1RM = () => (dispatch, getState) => {
     const results = OneRMCalculator.calculate1RM(exercise, tagsToInclude, tagsToExclude, daysRange, velocity, metric, allSets);
 
     // analytics
-    logCalculate1RMAnalytics(
-        state,
+
+    const calculateE1RMParams = analyticsParams(
```
Comment (JonnyLinja) [inline app/features/analysis/calculate/OneRMCalculateActions.js]: - [ ] analyticsParams

File: app/redux/reducers/AnalysisReducer.js [hunk 17]
```diff
@@ -177,6 +180,14 @@ const AnalysisReducer = (state = defaultState, action) => {
                 dragged: !state.dragged,
             };
         case CALC_1RM:
+            let e1RMCalcs = state.e1RMCalcs;
```
Comment (JonnyLinja) [inline app/redux/reducers/AnalysisReducer.js]: - [ ] oneRMAnalytics

File: app/redux/reducers/AnalysisReducer.js [hunk 18]
```diff
@@ -177,6 +180,14 @@ const AnalysisReducer = (state = defaultState, action) => {
...
+            let newCalc = action.e1RMCalc;
```
Comment (JonnyLinja) [inline app/redux/reducers/AnalysisReducer.js]: - [ ] currentOneRMAnalytics

File: app/redux/reducers/AnalysisReducer.js [hunk 19]
```diff
@@ -177,6 +180,14 @@ const AnalysisReducer = (state = defaultState, action) => {
...
+            
+            // filter out existing exercise + tags combos
+            let filteredCalcs = e1RMCalcs.filter((calc) => {
```
Comment (JonnyLinja) [inline app/redux/reducers/AnalysisReducer.js]: - [ ] I'd just do storedOneRMAnalytics = filter blah blah

File: app/features/analysis/calculate/OneRMCalculateActions.js (hunk 16 above)
Comment (Malekai) [reply app/features/analysis/calculate/OneRMCalculateActions.js]: I made it oneRMAnalyticsParams since there is already a function called analyticsParams

**Pull request merged**